nixos/common/detect-hostname-change.nix

```diff
@@ -13,13 +13,16 @@
 
   config = lib.mkIf (config.srvos.detect-hostname-change.enable && config.networking.hostName != "") {
     system.preSwitchChecks.detectHostnameChange = ''
+      actual=$(< /proc/sys/kernel/hostname)
+
       # Ignore if the system is getting installed
-      if [[ ! -e /run/current-system ]]; then
+      # https://github.com/nix-community/nixos-images/blob/2fc023e024c0a5e8e98ae94363dbf2962da10886/nix/installer.nix#L12-L13
+      if [[ ! -e /run/current-system || "$actual" == "nixos" || "$actual" == "nixos-installer" ]]; then
```

Comment:
Why are we including `nixos` here?

Reply:
By default the NixOS installer ISOs use `nixos` as the hostname

Reply:
That's not specific to the installer, but just the default for networking.hostName if unset.

So if your goal here is to check for installers, it's probably preferable to check /etc/os-release for VARIANT_ID as nixos-anywhere does. Or an other method which isn't as likely to yield false matches.

If it's even needed, the check is easy to disable during install via NIXOS_NO_CHECK

I proposed that for nixos-anywhere in https://github.com/nix-community/nixos-anywhere/pull/447

Reply:
We can't check `/etc/os-release` because this runs as part of activation which runs inside the `nixos-enter` `chroot`

Instead after some testing I realized that `/run/booted-system` won't exist when installing as it gets created by stage 2 normally

I've also removed the `nixos` hostname check